cluster-autoscaler/simulator/clustersnapshot/clustersnapshot.go

```diff
@@ -41,8 +41,6 @@ type ClusterSnapshot interface {
 	AddPod(pod *apiv1.Pod, nodeName string) error
 	// RemovePod removes pod from the snapshot.
 	RemovePod(namespace string, podName string, nodeName string) error
-	// IsPVCUsedByPods returns if the pvc is used by any pod, key = <namespace>/<pvc_name>
-	IsPVCUsedByPods(key string) bool
```

Comment:
There appear to be some other implementations of this interface and usages across the codebase (e.g., `cluster-autoscaler/simulator/clustersnapshot/basic.go`), sorry if those are in another commit that I didn't see! But  think we'll need to clean this up everywhere.

Reply:
This PR adapts both `ClusterSnapshot` implementations:
* basic: https://github.com/kubernetes/autoscaler/pull/7466/files#diff-29296088fba933fa837b2efdce93b2423b19247d99f59f1d8ef1bc8d5e8c6915
* delta: https://github.com/kubernetes/autoscaler/pull/7466/files#diff-e54b47f2d5a1526b9cde451164b5ab44aadd9f5a3ec5f93017ba4cd3a0faba05

Is there something missing?

Reply:
Why do we no longer need this? IIRC this was never directly used in CA, but we needed to be able to satisfy the scheduler NodeInfo lister interface. Is that no longer the case? I feel like I'm missing some important part of CA/scheduler integration

Reply:
After talking with Kuba - the functionality still exists and scheduler was accessing it via the lister interface anyway. This makes sense to me.

Reply:
That explanation satisfies me, thanks!

Reply:
Because of this point:

> We definitely still need to have this implemented and expose it to the scheduler framework - this is done via the StorageInfos() method.

The only redundant part was the `IsPVCUsedByPods` method being exposed directly on `ClusterSnapshot`, because it's already accessible via `ClusterSnapshot.StorageInfos().IsPVCUsedByPods`. `ClusterSnapshot.StorageInfos()` is definitely still needed for PVC integration with scheduler, so we don't want to drop the `IsPVCUsedByPods` implementations.

Reply:
True, I've missed that StorageInfos() returns self, so this is needed in implementation. LGTM then.